### PR TITLE
Add Application Agent

### PR DESCRIPTION
**Pull request by filariow**
Update the repo to also host the application agent's source code

Signed-off-by: Francesco Ilario <filario@redhat.com>

### CODE REVIEW

File: cmd/primaza/main.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "ClusterEnvironment")
+		os.Exit(1)
+	}
+	if err = (&controllers.ServiceBindingReconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "ServiceBinding")
```
Comment (dperaza4dustbit) [inline cmd/primaza/main.go:103]: We don't need ServiceBinding in primaza right?

**Review by dperaza4dustbit — COMMENTED** [review]
Looks good, just a couple of questions

File: cmd/agents/app/main.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	if err != nil {
+		setupLog.Error(err, "unable to start manager")
+		os.Exit(1)
+	}
+
+	if err = (&controllers.ServiceBindingReconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "ServiceBinding")
```
[the remark this answers is not in the document]
Comment (filariow) [reply cmd/agents/app/main.go:105]: yes, nice catch!

File: cmd/primaza/main.go (hunk 1 above)
Comment (filariow) [reply cmd/primaza/main.go:103]: you're right, we can remove this.

File: make/agents/app/build.mk [hunk 3]
```diff
@@ -0,0 +1,22 @@
+##@ Build
+
+AGENTSAPP_MAIN=./cmd/agents/app/main.go
+
+.PHONY: build
+build: fmt vet ## Build manager binary.
+	$(GO) build -o bin/agentapp ${AGENTSAPP_MAIN}
```
Comment (sadlerap) [inline make/agents/app/build.mk:7]: I'm not a huge fan of having multiple definitions of the same rule, e.g. a `build` target for both primaza and agents.  Could we unify these somehow?

File: make/agents/app/deploy.mk [hunk 4]
```diff
@@ -0,0 +1,14 @@
+KUSTOMIZE_ARGS=--load-restrictor LoadRestrictionsNone
+KUSTOMIZE_OVERLAY=config/crd/overlays/agentapp
+
+ifndef ignore-not-found
+  ignore-not-found = false
+endif
+
+.PHONY: install
+install: manifests kustomize ## Install CRDs into the K8s cluster specified in ~/.kube/config.
+	$(KUSTOMIZE) build $(KUSTOMIZE_ARGS) $(KUSTOMIZE_OVERLAY) | kubectl apply -f -
```
Comment (sadlerap) [inline make/agents/app/deploy.mk:10]: This doesn't do anything agent specific.  Are we intending on installing agent CRDs separately?

File: make/primaza/deploy.mk [hunk 5]
```diff
@@ -0,0 +1,23 @@
+##@ Deployment
+
+ifndef ignore-not-found
+  ignore-not-found = false
+endif
+
+.PHONY: install
+install: manifests kustomize ## Install CRDs into the K8s cluster specified in ~/.kube/config.
+	$(KUSTOMIZE) build config/crd | kubectl apply -f -
```
Comment (sadlerap) [inline make/primaza/deploy.mk:9]: Ditto; this doesn't do anything specific for primaza.

**Review by sadlerap — COMMENTED** [review]
I'll have more feedback on Monday, but from the outset, it looks pretty good so far.  I'm definitely not used to the style of make you're using here, however.

File: make/agents/app/deploy.mk (hunk 4 above)
Comment (filariow) [reply make/agents/app/deploy.mk:10]: agents will eventually install only a subset of the CRDs

File: make/agents/app/build.mk (hunk 3 above)
Comment (filariow) [reply make/agents/app/build.mk:7]: as decided with the team, we are going to try this approach for now. In future we may roll back to a more make-ish approach or even introduce a different build automation tool

**Review by dperaza4dustbit — DISMISSED** [review]
/lgtm
I say lets give this a try

**Review by sadlerap — APPROVED** [review]
LGTM

**Pull request merged**